plugins/inputs/chrony/chrony.go

```diff
@@ -38,6 +40,24 @@ func (*Chrony) SampleConfig() string {
 	return sampleConfig
 }
 
+// dialUnix opens an unixgram connection with chrony
+func dialUnix(address string) (*net.UnixConn, error) {
+	base, _ := path.Split(address)
+	local := path.Join(base, fmt.Sprintf("chronyc.%d.sock", os.Getpid()))
+	conn, err := net.DialUnix("unixgram",
+		&net.UnixAddr{Name: local, Net: "unixgram"},
+		&net.UnixAddr{Name: address, Net: "unixgram"},
+	)
```

Comment:
Why do you need to setup a local address? Why did you remove the timeout used before? Couldn't you let the OS figure out the local socket and use
```golang
net.DialTimeout("unixgram", u.Path, time.Duration(c.Timeout))
```
instead?

Reply:
Same as above, it doesn't work and fails with a -EAGAIN. Don't know why. The C client does the same